src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1096,5 +1102,41 @@ private static void ThrowMinMaxException<T>(T min, T max)
         {
             throw new ArgumentException(SR.Format(SR.Argument_MinMaxValue, min, max));
         }
+
+        public static double ScaleB(double x, int n)
+        {
+            double y = x;
+            if (n > 1023)
```

Comment:
Its probably not particularly important here, but the MSVC and the JIT tends to emit (assuming no PGO data):
```
if (condition) {
  // predicted
}
else {
  // unpredicted
}
```

This is inverse from what Clang/GCC do, so the `n > 1023` and `n < -1022` cases are going to be predicted as taken by default, while I believe they are edge cases in practice.

You can compare JIT (noting this is 3.1 output, not .NET 5, so there may be a couple minor differences): <https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGAHKAJYA3bBhgNIAO1wYGAEwgBXYABtxAZRYBBADIBRAEIB9FuQYBeBgA4AdAE4rVlEgCsSAOwo65ci6t6AajoABncAbgYAekiGYIRyXnJg0jouHn5hUXEpGXklVQ1tfWMWUgsGUhtSUhdQuis/WvdSYPIUPTgQqwjo2PjeOCTqtO4MkTEJCGlZBWU1Bk1dQxNGSztg0PI7OxqUT2Dt0h6YuISXVOoRpkZiciQ8uY0wbDUDAApZgoYENAYBSVkkgAlFdONQeBCHl8AJ7lBBhK4QgQAMwYb0kDAAfAwhnQQeDIYSwYTCbCAFSWRbFEzkBEEkk8DFwSy4ukMpGo9FYnHJPGIknE9mkhgUhZFZamNlCyFMlm8qXSngotEY7G4/GKiGCzUy8qs/lCgC+BshxvpppNMBUuHEyq5AB4GINkqQNQKTRDyZTxSUymSxUsSnQFeyMQE5dUnQxziGGXaMY7ndU3ULtZqvQHqaURZmJcGPSSwxGynBo/nzYr4wxE0NXQWifXQ+Uk0dGxCzYqOySzSbPvNFOUDAIMCwpkIYFAxFAbABJAGoIcYXAAFQgABF8mo3tuVFMAOZAt5xOjI1EBBjA6uOlyuoGxnjEdwMckMRQhs2GoA>

To MSVC and Clang: <https://godbolt.org/z/MEYGcz>

Reply:
This is good to know. I tried reworking the logic to take this into consideration, roughly something like:

```
if (n <= 127 && n >= -126) {
    // n is in expected range, just compute and return result
} else {
    if (n > 127) {
        // Make adjustments
    }
    else if (n < -126) {
        // Make adjustments
    }
}
```

But I'm not seeing an improvement in perf. I can push if you'd like to take a look, but not sure it's worth it.

Reply:
> I can push if you'd like to take a look, but not sure it's worth it.

Right, I don't think in this particular case it is particular import, I just wanted to call it out as it might be relevant if additional work is done in this or related areas later.

Reply:
I was thinking that if those edge cases were moved into their own method, this could become inlineable. But it doesn't seem to get inlined even then.